### PR TITLE
Wired up UD and ENS Resolution to UI

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Wired up UnstoppableDomains and ENS Resolution to UI

1) Refactored/Moved `Send` methods into a reusable hook `useSend`, will also clear `toAddress` and `sendAmount` after creating transaction.
2) If a url ends with `.eth` it will call the `ensGetEthAddr` method, if the url is registered it will return an address that we will use to send, if the url is not registered will return a `Domain <name> is not registered`  error.
2) If a url ends with `.crypto` it will call the `unstoppableDomainsGetEthAddr` method, if the url is registered it will return an address that we will use to send, if the url is not registered will return a `Domain <name> is not registered`  error.
3) If the address starts with `0x`, it will check if its a valid address, if not it will return a `Not a valid ETH address` error
4) If the input value is not recognized it will return a `Not a valid address` error

Resolves <https://github.com/brave/brave-browser/issues/18681>
Resolves <https://github.com/brave/brave-browser/issues/18656>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/137414185-9aa59b11-2225-42d8-b5fd-f2f1133d52fb.mov

### CODE REVIEW

File: components/brave_wallet_ui/common/hooks/send.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { GetEthAddrReturnInfo } from '../../constants/types'
+import { isValidAddress } from '../../utils/check-eth-address'
```
Comment (onyb) [inline components/brave_wallet_ui/common/hooks/send.ts:15]: Can we rename this file to `address-utils`?

File: components/brave_wallet_ui/common/hooks/send.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  const onSetSendAmount = (value: string) => {
+    setSendAmount(value)
+  }
+
+  const setNotRegisteredError = (url: string) => {
+    setAddressError(`${getLocale('braveWalletNotDomainFirst')} ${url} ${getLocale('braveWalletNotDomainSecond')}`)
+  }
+
+  React.useMemo(() => {
```
Comment (onyb) [inline components/brave_wallet_ui/common/hooks/send.ts:53]: We could probably do a `useEffect` here, right?

File: components/brave_wallet_ui/common/hooks/send.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        setNotRegisteredError(valueToLowerCase)
+      }).catch(e => console.log(e))
+      return
+    }
+
+    // If value starts with 0x, will check if it's a valid address
+    if (valueToLowerCase.startsWith('0x')) {
+      setToAddress(toAddressOrUrl)
+      isValidAddress(toAddressOrUrl, 20) ? setAddressError('') : setAddressError(getLocale('braveWalletNotValidEthAddress'))
```
Comment (onyb) [inline components/brave_wallet_ui/common/hooks/send.ts:87]: Line too long? You could do:

```ts
isValidAddress(toAddressOrUrl, 20)
  ? setAddressError('')
  : setAddressError(getLocale('braveWalletNotValidEthAddress'))
```

File: components/resources/wallet_strings.grdp [hunk 4]
```diff
@@ -291,6 +292,10 @@
     <message name="IDS_BRAVE_WALLET_EDIT_GAS_GWEI" desc="Edit gas panel gwei label">Gwei</message>
     <message name="IDS_BRAVE_WALLET_EDIT_GAS_SET_CUSTOM" desc="Edit gas panel set custom button">Set custom</message>
     <message name="IDS_BRAVE_WALLET_EDIT_GAS_SET_SUGGESTED" desc="Edit gas panel set suggested button">Set suggested</message>
+    <message name="IDS_BRAVE_WALLET_NOT_VALID_ETH_ADDRESS" desc="Send input not valid ETH address error">Not a valid ETH address</message>
+    <message name="IDS_BRAVE_WALLET_NOT_VALID_ADDRESS" desc="Send input not valid address error">Not a valid address</message>
+    <message name="IDS_BRAVE_WALLET_NOT_DOMAIN_FIRST" desc="Send input not registered domain first error">Domain</message>
+    <message name="IDS_BRAVE_WALLET_NOT_DOMAIN_SECOND" desc="Send input not registered domain second error">is not registered</message>
```
Comment (onyb) [inline components/resources/wallet_strings.grdp:298]: You can do the following:

```
Domain <ph name="VALUE">$1</ph> is not registered
```

and in the React components, do `getLocale('braveWalletNotDomain').replace('$1', url)`.

File: components/brave_wallet_ui/common/hooks/send.ts (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/send.ts]: Re-formated!

File: components/resources/wallet_strings.grdp (hunk 4 above)
Comment (Douglashdaniel) [reply components/resources/wallet_strings.grdp]: Changed! Works nice!

**Review by yrliou — APPROVED** [review]
LGTM

**Pull request merged**